File: leetcode_py/leetcode_03.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):

        last = [-1] * 256

        left = 0
        ans = 0

        for right in range(len(s)):

            if last[ord(s[right])] != -1:
                left = max(left, last[ord(s[right])] + 1)

            last[ord(s[right])] = right

            ans = max(ans, right - left + 1)

        return ans

File: leetcode_practice/leetcode_03.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int>last(256,-1);
        int n=s.size();
        int left =0,ans=0;
        for(int r=0;r<n;r++){
            if(last[s[r]]!=-1){
                left=max(left,last[s[r]]+1);
            }
            last[s[r]]=r;
            ans=max(ans,r-left+1);
        }
        return ans;
    }
};
